docs/pages/cardinal/ecs.mdx

```diff
@@ -0,0 +1,42 @@
+# Introduction to ECS
+
+Cardinal follows the ECS (Entity-Component-System) architecture which facilitates fast and efficient updates to game
+state, as well as enabling painless extensibility to your games.
+
+ECS, or Entity-Component-System, stands as a programming paradigm specifically tailored for game development. While ECS
+itself is conceptually straightforward, its effective implementation demands a nuanced understanding. Familiarizing
+yourself with the ECS model is crucial before delving into the capabilities of Cardinal.
+
+
+### Entities
+
+Entities are simply a unique identifier of something in the game. Entities can be players, objects, NPCs, etc. They are
+the simplest unit in the ECS system.
+
+### Components
+
+Components are data fragments that bind to entities, offering descriptive attributes. Imagine a "Health" component for
+players, tracking their health status. Maintaining component generality is key – for instance, a health component can be
+reused for different types of entities, such as animals or destructible objects.
+
+### Entities, part two
+
+Now that we've explained components, let's revisit entities. In the ECS realm, entities aren't conjured from thin air;
+they are composed through the combination of one or more components. Consider a fighting game, where a player is
+composed of both a "Health" component, quantifying hit points, and an "Attack" component, dictating the inflicted damage
+potential.
+
+### Systems
+
+Systems are the heartbeat of ECS, encompassing functions that manipulate entities, allowing your game logic comes to
```

Comment:
"allowing your game logic comes to come to life."

"allowing your game logic to come to life."